src/test/python/test_signer.py

```diff
@@ -0,0 +1,188 @@
+#   Copyright 2017 Covata Limited or its affiliates
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import pytest
+from covata.delta import signer
+import json
+import re
+from datetime import datetime
+import six.moves.urllib as urllib
+from covata.delta.signer import __encode_uri, SignatureMaterial
+from freezegun import freeze_time
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import padding
+from covata.delta import crypto
+
+freeze_time("2017-01-03 14:48:10")
+
+UNDESIRED_HEADERS = ["Connection", "Content-Length"]
+SIGNING_ALGORITHM = "CVT1-RSA4096-SHA256"
+CVT_DATE_FORMAT = "%Y%m%dT%H%M%SZ"
+
+
```

Comment:
Can you add a test for when the resource path is "/"?

See line 160: https://coveralls.io/builds/10338869/source?filename=src%2Fmain%2Fpython%2Fcovata%2Fdelta%2Fsigner.py